Makefile

```diff
@@ -0,0 +1,31 @@
+# define the name of the virtual environment directory
+VENV := venv
+
+# default target, when make executed without arguments
+all: venv
+
+$(VENV)/bin/activate: requirements.txt
+	python3 -m venv $(VENV)
+	./$(VENV)/bin/pip install -r requirements.txt
+	./$(VENV)/bin/pip install pandas
+
+# venv is a shortcut target
+venv: $(VENV)/bin/activate
+
+clean:
+	rm -rf $(VENV)
+	find . -type f -name '*.pyc' -delete
+
+run-clickhouse:
+	@docker run -p 9000:9000 --ulimit nofile=262144:262144 -it clickhouse/clickhouse-server
```

Comment:
`--ulimit nofile=262144:262144` what for?

Reply:
it's a recommendation of ClickHouse developers.

you can see an official description of image:
https://hub.docker.com/r/clickhouse/clickhouse-server